examples/trk_subsampler.rs

```diff
@@ -42,15 +44,37 @@ fn main() {
 
     if let Ok(percent) = args.get_str("--percent").parse::<f32>() {
         let percent = percent / 100.0;
-        let mut rng = rand::thread_rng();
+        let mut rng = rand::weak_rng();
 
         for streamline in reader.into_iter() {
             if rng.gen::<f32>() < percent {
                 writer.write(&streamline);
             }
         }
-    } else if let Ok(_nb) = args.get_str("--number").parse::<usize>() {
-        panic!("Not implemented yet");
+    } else if let Ok(nb) = args.get_str("--number").parse::<usize>() {
+        if let Ok(seed) = args.get_str("--seed").parse::<u8>() {
+            let size = reader.header.nb_streamlines;
+            let mut rng = rand::XorShiftRng::from_seed([seed; 16]);
+
+            let mut sampled_indices = rand::seq::sample_indices(&mut rng, size, nb);
+            sampled_indices.sort();
+
+            let mut iter_indices = sampled_indices.into_iter();
+            let mut index = iter_indices.next();
+
+            for (idx, streamline) in reader.into_iter().enumerate() {
+                if let Some(index_to_save) = index {
+                    if idx == index_to_save {
+                        writer.write(&streamline);
+                        index = iter_indices.next();
+                    }
+                } else {
+                    break;
```

Comment:
I have another suggestion for this part. You could iterate on `iter_indices` instead and [skip](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.nth) most iterations on the reader. Something like:

    let mut reader_iter = reader.into_iter();
    let mut last = 0;
    for idx in indices {
        let streamline = reader_iter.nth(idx - last).unwrap();
        writer.write(streamline ); // Could be a one-liner too. Maybe missing a &
        last = idx;
    }

I don't like `last`, but there's no no condition now! What do you think?

Reply:
I think I need to read more carefully the documentation to see these kinds of life savor `nth`! I like it.